### PR TITLE
Feature/precipitation and historical data

### PR DESCRIPTION
**Pull request by ZachMarston**
# Added Functionality
## What?
Added precipitation data and some changes to how historical data is gathered
## Why?
Because I needed to for weather alerts.
## PR Checklist
- [x ] Merged latest master
- [x] Updated version number
## Breaking Changes
I didn't find any

### CODE REVIEW

File: sentera/weather.py [hunk 1]
```diff
@@ -225,8 +226,23 @@ def split_time_interval(time_interval, weather_type, weather_interval):
 
         return time_intervals
 
-    elif weather_type == WeatherType.Historical:
-        return [["01-01", "07-01"], ["07-01", "12-31"]]
+    if weather_type == WeatherType.Historical:
+        if not time_interval:
+            raise ValueError("Time interval needed for historical weather types")
+        try:
+            start = datetime.datetime.strptime(time_interval[0], "%m/%d")
```
Comment (ZachMarston) [inline sentera/weather.py:233]: This doesn't work for 02/29 and DTN is working on a fix. We will need to have this changed before 2024.

File: sentera/weather.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 import aiohttp
 import pandas as pd
 import tqdm
-from pandas.io.json import json_normalize
+from pandas import json_normalize
```
Comment (Olson3R) [inline sentera/weather.py:21]: This should get rid of a warning I believe 😄

File: sentera/weather.py [hunk 3]
```diff
@@ -226,7 +227,22 @@ def split_time_interval(time_interval, weather_type, weather_interval):
...
+        except ValueError:
+            raise ValueError("Incorrect time interval format, should be MM/DD")
+
+        try:
+            end = datetime.datetime.strptime(time_interval[1], "%m/%d")
+        except ValueError:
+            raise ValueError("Incorrect time interval format, shoule be MM/DD")
+        diff = (end - start).days
+        if diff >= 0:
+            return [[time_interval[0], time_interval[1]]]
```
Comment (Olson3R) [inline sentera/weather.py:243]: Can/should this be?
```suggestion
            return [time_interval]
```

File: sentera/weather.py [hunk 4]
```diff
@@ -226,7 +227,22 @@ def split_time_interval(time_interval, weather_type, weather_interval):
...
+        else:
+            return [[time_interval[0], "12-31"], ["01-01", time_interval[1]]]
```
Comment (Olson3R) [inline sentera/weather.py:245]: Does this path handle periods where something crosses new years?

File: sentera/weather.py [hunk 5]
```diff
@@ -226,7 +227,22 @@ def split_time_interval(time_interval, weather_type, weather_interval):
         return time_intervals
 
     elif weather_type == WeatherType.Historical:
-        return [["01-01", "07-01"], ["07-01", "12-31"]]
+        if not time_interval:
+            raise ValueError("Time interval needed for historical weather types")
+        try:
+            start = datetime.datetime.strptime(time_interval[0], "%m/%d")
+        except ValueError:
+            raise ValueError("Incorrect time interval format, should be MM/DD")
```
Comment (Olson3R) [inline sentera/weather.py:235]: This seems like a good candidate for a helper function to format/verify DTN dates.

File: sentera/weather.py (hunk 4 above)
Comment (ZachMarston) [reply sentera/weather.py:245]: That is exactly what it was intended to take care of :)

File: sentera/weather.py (hunk 3 above)
Comment (ZachMarston) [reply sentera/weather.py:243]: I tried to keep it consistent with the format that was used earlier.

File: sentera/weather.py (hunk 2 above)
Comment (ZachMarston) [reply sentera/weather.py:21]: Yes it does. I was getting sick of seeing that warning.

File: sentera/weather.py [hunk 6]
```diff
@@ -171,6 +172,37 @@ def create_params(weather_type, time_interval):
...
 
+WEATHER_STRINGS = {
+    WeatherType.Recent: {"format": "%Y/%m/%d", "date_string": "YYYY/MM/DD"},
+    WeatherType.Historical: {"format": "%m/%d", "date_string": "MM/DD"},
+}
+
+
+def check_time_interval(time_interval, weather_type):
+    """
+        Cehck the time interval.
```
Comment (cliff-dellamano) [inline sentera/weather.py:183]:
```suggestion
        Check the time interval.
```

File: sentera/weather.py [hunk 7]
```diff
@@ -171,6 +172,37 @@ def create_params(weather_type, time_interval):
...
+
+    :param time_interval:
+    :param weather_type:
+    :return:
+    """
+    if not time_interval:
+        raise ValueError("Time interval needed for recent weather types")
```
Comment (cliff-dellamano) [inline sentera/weather.py:190]:
```suggestion
        raise ValueError(f"Time interval needed for {weather_type} weather types")
```

Comment (cliff-dellamano) [reply sentera/weather.py]: You could also do this with basic string concatenation if you don't want to use the f string.

**Pull request merged**